core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -136,6 +138,18 @@ public static void dropTableData(FileIO io, TableMetadata metadata) {
     deleteFile(io, metadata.metadataFileLocation(), "metadata");
   }
 
+  /**
+   * Drops view metadata files referenced by ViewMetadata.
+   *
+   * <p>This should be called by dropView implementations
+   *
+   * @param io a FileIO to use for deletes
+   * @param metadata the last valid ViewMetadata instance for a dropped view.
+   */
+  public static void dropViewMetadata(FileIO io, ViewMetadata metadata) {
```

Comment:
This change can go as part of Hive-View implementation.

Reply:
from what I understand, this PR only refactors a bunch of things to make it easier to add view support. That means we probably shouldn't be adding view-specific things in this PR as we just want to move stuff around and provide the exact same code behavior that previously existed, right?

Reply:
this hasn't been addressed yet

Reply:
My Bad, I missed to push the corresponding commit.